### ISSUE TITLE
Vario not working with FS-CAT01 on TX16S

### ISSUE THREAD

**Issue by sukhoi222** [EdgeTX/edgetx#1948]
I have tried a lot of Edgetx updates, but I wasn´t able to use vario sensor FS-CAT01 on my Radiomaster.
I read Vspd needs to be calculated, because the self sensor, don´t do it. I also read that this issue is solved by NV14, but I can´t not make it work on my TX16S. I looked for scripts, but the ones I found don´t work with version 2.5, 2.6 and 2.7. 
Pls help.

**Comment by pfeerick** [comment]
I think for this to work since you're using MPM on the TX16S rather than the internal RF of the NV14 (which is what #1485 implemented the VSpd calculated sensor for) that we also need to implement it for https://github.com/EdgeTX/edgetx/blob/main/radio/src/telemetry/flysky_ibus.cpp so that the VSpd calculated sensor is generated with the MPM also.

**Comment by sukhoi222** [comment]
Good morning,
I appreciate very much you help and explanation. Unfortunately I am not good at programming and don't know what to do with the programming text you attached with the explanation.
As far as I understand, you managed to solve Vspd for Radiomaster. Is that right?
How should I proceed to make it work in my Radiodimaster?

**Comment by pfeerick** [comment]
No, it was done for the NV14 and future FlySky radios. I'm just suggesting either to myself or whoever picks this up that that is probably where to start looking.

**Comment by zashybys** [comment]
There is such a lua script _FlySky Variometer_ for **flysky Rx** + FS-CAT01 Alt sensor  https://hackaday.io/project/175220-flysky-variometer but it works on OpenTX only. Because for some strange reason EdgeTX doesn't have a CUSTOM SCRIPTS tab at all. Could you return the CUSTOM SCRIPTS tab to EdgeTX as it is in OpenTX?

Could you somehow fix the vario in EdgeTX to work not only from the VSpd sensor but also from the Alt, similar to what is realized in the FS-i6 Tx firmware https://github.com/qba667/FlySkyI6/wiki/Vario vario works good enough and from the Alt sensor.

Or maybe someone could rewrite the same _FlySky Variometer_ script but customized to EdgeTX 2.7+

Or it would be best to be able to create a calculated VSpd sensor from a real Alt sensor. But this requires an additional formula such as the instantaneous change in height over time, which is actualy VSpd.

I use a TX16S transmitter and I am not a programmer. If I can't find a solution I'll have to go back to OpenTX just for this script only. Because I fly rc gliders mostly and I already have 4 flysky Rx and a couple of FS-CAT01 and couple of voltage sensors FS-CVT01.

**Comment by sukhoi222** [comment]
I'm interested too. I'm eaiting for long time for this update in EdgeTx. 
I used former script for Opentx, I worked fine. But now with Edgetx seems that I went some steps back, loosing performances. Anyone ones, whether in 2.8 update, that will be already solved?
Thanks. 😃
